### PR TITLE
FEAT: Add new parser XFSQuotaState

### PR DESCRIPTION
**Pull request by TZ3070**
### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:

Add new parser XFSQuotaStatue for adviser rule

### CODE REVIEW

File: insights/parsers/xfs_quota.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        3
+        >>> quota_state.user_quota[0] == {'device': '/dev/sdd', 'accounting': 'ON', 'enforcement': 'ON', 'blocks_grace_time': '7 days', 'blocks_max_warnings': '5', 'inodes_grace_time': '7 days', 'inodes_max_warnings': '5', 'realtime_blocks_grace_time': '7 days'}
+        True
+    """
+
+    def parse_content(self, content):
+        self.user_quota = []
+        self.group_quota = []
+        data = None
+        for line in get_active_lines(content):
```
Comment (xiangce) [inline insights/parsers/xfs_quota.py:80]: As currently, performance is considered in high priority, it's not recommended to use the `get_active_lines` against `content`, especially if it's got from a command (it's assumed that there are no comment lines, e.g. `# xxx`. Better skip the empty lines in the following loop by hand, e.g. `if not line.strip(); continue`

File: insights/parsers/xfs_quota.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+from insights.core import Parser
+from insights.core.exceptions import SkipComponent
+from insights.core.plugins import parser
+from insights.parsers import get_active_lines
+from insights.specs import Specs
+
+
+@parser(Specs.xfs_quota_statue)
+class XFSQuotaStatue(Parser):
```
Comment (xiangce) [inline insights/parsers/xfs_quota.py:19]: CommandParser?

File: insights/specs/default.py [hunk 3]
```diff
@@ -707,6 +707,7 @@ class DefaultSpecs(Specs):
     x86_ibrs_enabled = simple_file("sys/kernel/debug/x86/ibrs_enabled")
     x86_pti_enabled = simple_file("sys/kernel/debug/x86/pti_enabled")
     x86_retp_enabled = simple_file("sys/kernel/debug/x86/retp_enabled")
+    xfs_quota_statue = simple_command("/sbin/xfs_quota -x -c 'state -gu'")
```
Comment (xiangce) [inline insights/specs/default.py:710]: You also need to add its `simple_file` to the `insights_archive.py` if it also can be collected via uploader.json

File: insights/parsers/xfs_quota.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
+"""
+xfs_quota Commands
+==================
+
+Parser contains in this module is:
+
+XFSQuotaStatue - command ``/sbin/xfs_quota -x -c 'state -gu'``
```
Comment (xiangce) [inline insights/parsers/xfs_quota.py:7]: `XFSQuotaState`?

**Review by xiangce — CHANGES_REQUESTED** [review]

File: insights/parsers/xfs_quota.py (hunk 4 above)
Comment (TZ3070) [reply insights/parsers/xfs_quota.py]: @xiangce Updated, please review again

**Review by xiangce — APPROVED** [review]

**Pull request merged**